spring-integration-feed/src/main/java/org/springframework/integration/feed/inbound/FeedEntryMessageSource.java

```diff
@@ -258,7 +262,7 @@ private SyndFeed buildSyndFeed() throws IOException, URISyntaxException, Interru
 			HttpRequest request =
 					HttpRequest.newBuilder()
 							.GET()
-							.uri(this.feedUrl.toURI())
+							.uri(Objects.requireNonNull(this.feedUrl).toURI())
```

Comment:
Well, this one is indeed can be `null` according to the constructors logic.
However, since we have here `if..else`, there is no chance that we come at this point and it is still null.
So, this method has to be marked with a `@SuppressWarnings("NullAway") // Dataflow analysis limitation` instead.